src/routes.rs

```diff
@@ -166,7 +192,16 @@ pub async fn dispense_tokens(
             wallet
                 .get_spendable_resources(AssetId::BASE, THE_BIGGEST_AMOUNT)
                 .await
-                .map_err(|e| error(format!("Failed to get resources: {}", e)))?
+                .map_err(|e| {
+                    dispense_tracker
+                        .lock()
+                        .unwrap()
+                        .remove_in_progress(&address);
+                    error(
```

Comment:
I meant we need to call `remove_in_progress` on any error. You can move calling of the `remove_in_progress` into `error` function, then you should cover all cases=)